crates/storage/src/db.rs

```diff
@@ -0,0 +1,28 @@
+use crate::redb_interface::ReamDB;
+use crate::errors::StoreError;
+
+pub struct Database{
+    /// Database for the ream client 
+    /// 
+    /// Currently this implementation has a single reference to redb instance
+    /// but eventually we would like to add support for multiple backends and 
+    /// can be set behind feature flags
+    ///
+    /// `redb::Database` wrapper around with async security features
+    db: ReamDB,
```

Comment:
```suggestion
    pub db: ReamDB,
```

without pub this is unusable

Reply:
This is not resolved. Make sure you mark comments resolved only when it is.